external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -205,6 +205,28 @@ enum TypeAnnotation {
     Expression,
 }
 
+//**************************************************************************************************
+// Resolvable Module
+//**************************************************************************************************
+
+/// Trait to abstract over module-like structures for member resolution
+trait ResolvableModule {
+    /// Returns iterator over structs: (name, type_params_len, field_info, loc)
+    fn structs(&self) -> impl Iterator<Item = (DatatypeName, usize, FieldInfo, Loc)>;
+
+    /// Returns iterator over enums: (name, type_params_len, loc, variants)
+    /// where variants is Vec of (variant_name, field_info, variant_loc)
+    fn enums(
+        &self,
+    ) -> impl Iterator<Item = (DatatypeName, usize, Loc, Vec<(VariantName, FieldInfo, Loc)>)>;
+
+    /// Returns iterator over functions: (name, type_params_len, params_len)
+    fn functions(&self) -> impl Iterator<Item = (FunctionName, usize, usize)>;
+
+    /// Returns iterator over constants: (name, defined_loc)
+    fn constants(&self) -> impl Iterator<Item = (ConstantName, Loc)>;
+}
+
```

Comment:
Nit: I would consider moving this to its own file.

Reply:
Done!